wd_helper.au3

```diff
@@ -981,28 +981,63 @@ EndFunc   ;==>_WD_ElementOptionSelect
 ;                  |SELECTEDINDEX  - Retrieves 0-based index of the first selected <option> element
 ;                  |SELECTEDLABELS - Retrieves labels of selected <option> elements as 1D array
 ;                  |SELECTEDOPTIONS- Retrieves selected <option> elements as 2D array
+;                  |SINGLESELECT   - Select <option> element given as string and deselect all others
 ;                  |VALUE          - Retrieves value of the first selected <option> element
-;                  $aParameters    - [optional] List of parameters (depending on chosen $sCommand)
+;                  $vParameters    - [optional] List of parameters (depending on chosen $sCommand)
 ; Return values .: Success - Requested data returned by web driver.
 ;                  Failure - "" (empty string) and sets @error to one of the following values:
 ;                  - $_WD_ERROR_NoMatch
 ;                  - $_WD_ERROR_Exception
 ;                  - $_WD_ERROR_InvalidDataType
 ;                  - $_WD_ERROR_InvalidExpression
 ;                  - $_WD_ERROR_InvalidArgue
+;                  - $_WD_ERROR_ElementIssue
+;                  - $_WD_ERROR_NoMatch
 ; Author ........: Danp2
 ; Modified ......: mLipok
-; Remarks .......: If no option is selected, SELECTEDINDEX will return -1
+; Remarks .......: If no option is selected, SELECTEDINDEX will return -1.
 ; Related .......: _WD_FindElement, _WD_ExecuteScript
 ; Link ..........:
 ; Example .......: No
 ; ===============================================================================================================================
-Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand, $aParameters = Default)
+Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand, $vParameters = Default)
 	Local Const $sFuncName = "_WD_ElementSelectAction"
 	Local $vResult, $sScript
-
 	Local $sNodeName = _WD_ElementAction($sSession, $sSelectElement, 'property', 'nodeName')
 	Local $iErr = @error, $iExt = 0
+	Local Static $sScript_MultiSelectTemplate = StringReplace( _ ; it is declared as static to optimize AutoIt processing speed - this line will be processed once per script run
+			"function MultiSelectOption(SelectElement, LabelsToSelect, DeselectingNonListedLabels) {" & _
+			"	if ((LabelsToSelect.length > 1) && (SelectElement.multiple == false)) {" & _
+			"		return '';" & _
+			"	}; " & _
+			"	var options = SelectElement.options;" & _
+			"	var result = false;" & _
+			"	var waschanged = false;" & _
+			"	for (var i = 0, o; i < options.length; i++) {" & _
+			"		o = options[i];" & _
+			"		if (		(LabelsToSelect.indexOf(o.label)!= -1)" & _
+			"				&&	(o.disabled==false	&& (!(o.parentNode.nodeName == 'OPTGROUP' && o.parentNode.disabled)))" & _
+			"				&&	(o.hidden==false	&& (!(o.parentNode.nodeName == 'OPTGROUP' && o.parentNode.hidden)))" & _
+			"			) {" & _
```

Comment:
I wonder if we should set `$_WD_ERROR_ElementIssue` when any of `<option>` is `disabled` or `hidden` instead of just ignoring them. 

btw.
Currently if you use `SINGLESELECT` on such `<option>` then `$_WD_ERROR_NoMatch` is set, instead setting `$_WD_ERROR_ElementIssue`